src/main/java/zmq/ZMQ.java

```diff
@@ -822,14 +822,23 @@ public Selector get()
             return selector;
         }
 
+        void close()
+        {
+            if (selector != null) {
+                try {
+                    selector.close();
+                    selector = null;
+                }
+                catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+
         @Override
         public void finalize()
         {
```

Comment:
ATM, this is one of the place of the memory leaks. The code should never rely on finalize() being called to release the resource.